internal/platform/vendordetector.go

```diff
@@ -0,0 +1,193 @@
+package platform
+
+import (
+	stderrors "errors"
+	"fmt"
+
+	"github.com/jaypipes/ghw"
+	"github.com/openshift/dpu-operator/internal/daemon/plugin"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kind/pkg/errors"
+)
+
+type PlatformInfoProvider interface {
+}
+
+type VendorDetector interface {
+	IsDpuPlatform(platform Platform) (bool, error)
+	VspPlugin(dpuMode bool, vspImages map[string]string, client client.Client) (*plugin.GrpcPlugin, error)
+	IsDPU(pci ghw.PCIDevice) (bool, error)
+	GetVendorName() string
+}
+
+type PlatformInfo struct {
+	p         Platform
+	Detectors []VendorDetector
+}
+
+func NewPlatformInfo(p Platform) *PlatformInfo {
+	return &PlatformInfo{
+		p: p,
+		Detectors: []VendorDetector{
+			NewIntelDetector(),
+			NewMarvellDetector(),
+			// add more detectors here
+		},
+	}
+}
+
+func (pi *PlatformInfo) NewVspPlugin(dpuMode bool, vspImages map[string]string, client client.Client) (*plugin.GrpcPlugin, error) {
+	var detector VendorDetector
+	var err error
+
+	if dpuMode {
+		detector, err = pi.detectDpuPlatform(true)
+	} else {
+		detector, err = pi.detectDpuSystem(true)
+	}
+	if err != nil {
+		return nil, err
+	}
+	vspPlugin, err := detector.VspPlugin(dpuMode, vspImages, client)
+	if err != nil {
+		return nil, errors.Errorf("Error encountered when deploying VspPlugin: %v", err)
+	}
+
+	return vspPlugin, nil
+}
+
+func (pi *PlatformInfo) Getvendorname() (string, error) {
```

Comment:
Function name should be GetVendorName following convention.

Reply:
Agree, this is just a move still. Next PR will rework vendor detection completely